### ISSUE TITLE
Phoenix Router with auto-alias

### ISSUE THREAD

**Issue by sodapopcan** [elixir-lsp/elixir-ls#1013]
### Precheck

> Do a quick search and make sure the issue has not yet been reported

I did do a quick search for "router" and "route" in both open and closed issues.

### Environment

N/A since this is an observed behaviour based off of forum feedback.

### Current behavior

Hello ElixirLS team! ❤️

You are probably well aware of this but I couldn't find a logged issue so I'm opening one up.

See [here](https://elixirforum.com/t/issues-while-adding-hello-messenger-router/59333) for the latest example.  If required, I can dig up two other recent discussions I was involved in where newcommers had the same issue (I did a search to find them but apparently I talk too much on that site).

### Expected behavior

I don't particularly think this is ElixirLS' fault but I still wanted to log it.

The expected behaviour would be to expand a macro to see what it does before inserting an alias.  I can't say if that is unreasonable or not as I know nothing about language servers.  I don't even know if anything can be done, I just wanted to log it!

Thank you for all your work!  ❤️

**Comment by lukaszsamson** [comment]
For reference here’s another issue with router macros messing up aliases https://github.com/elixir-lsp/elixir-ls/issues/618. I’m more inclined to say this is phoenix design issue that manifest in ElixirLS. So far we only try to expand use macro and even doing so we do not execute any logic, simply insert code from _using_. I’m affraid expanding router macros would equal to compiling the file. That’s going beyond what AST can. This issue tracks elixir progres on the topic https://github.com/elixir-lang/elixir/issues/12645.

BTW all completions that insert alias are marked with alias string.

BTW2 we need a dedicated phoenix LS

**Comment by sodapopcan** [comment]
Makes sense!  Sorry for the dup, I likely bypassed it after reading "go to definition."

**Comment by lukaszsamson** [comment]
@sodapopcan is it still an issue in the current version? Improvements to complete and definition providers in Phoenix.Router landed in 0.19 and 0.20
